core/src/scss/utilities/mixins/link/_link-icon.scss

```diff
@@ -9,37 +9,83 @@
 /// @param {string} $vertical  - Adjustment of vertical position of icon, e.g., '2px' moves icon up 2 pixels, '-0.3em' moves it down 0.3 em.
+/// @param {string} $location - Location of icon, e.g., before or after.
 ///
 /// @group mixin
-@mixin link-icon($icon, $width: 0.65em, $animate: right, $vertical: null) {
+@mixin link-icon(
+  $icon,
+  $width: 0.65em,
+  $animate: right,
+  $vertical: null,
+  $location: after
+) {
   text-decoration: none;
   @supports (mask-repeat: no-repeat) {
-    &::after {
-      @include margin(null 0.3em $vertical 0.4em);
+    // Common properties for both before and after locations
+    &::#{$location} {
       @include size($width);
       display: inline-block;
       content: '';
-      mask: url("#{$su-image-path}/#{$icon}.svg") no-repeat 0 0;
+      mask: url('#{$su-image-path}/#{$icon}.svg') no-repeat 0 0;
       mask-size: contain;
-      background-color: color(link);
-      transition: transform 0.2s ease-in-out, mask-image 0.2s ease-in-out;
     }
 
     &:hover,
     &:focus {
-      &::after {
-        background-color: color(link--hover);
-        @if $animate == 'right' or $animate == 'true' {
-          transform: translateX(0.2em);
-        }
-        @else if $animate == 'down' {
+      &::#{$location} {
+        @if $animate == 'down' {
           transform: translateY(0.2em);
         }
         @else if $animate == 'up' {
           transform: translateY(-0.2em);
         }
-        @else if $animate == 'topright' {
-          transform: translate3d(0.15em, -0.15em, 0);
+      }
+    }
+
+    // Icon $location after link text
+    @if $location == 'after' {
+      &::after {
+        @include margin(null 0.1em $vertical 0.1em);
+        background-color: color(link);
+
+        @if $animate == 'right' or $animate == 'true' or $animate == 'topright' {
+          @include margin(null 0.3em $vertical 0.4em);
+          transition: transform 0.2s ease-in-out, mask-image 0.2s ease-in-out;
+        }
+      }
+
+      &:hover,
+      &:focus {
+        &::after {
+          background-color: color(link--hover);
+
+          @if $animate == 'right' or $animate == 'true' {
+            transform: translateX(0.2em);
+          }
+          @else if $animate == 'topright' {
+            transform: translate3d(0.15em, -0.15em, 0);
+          }
+        }
+      }
+    }
+
+    // If icon $location is not after then it'll be before!
+    @else {
+      &::before {
+        @include margin(null 0.1em $vertical 0.1em);
+        @if $animate == 'left' or $animate == 'true' {
+          @include margin(null 0.4em $vertical 0.3em);
+          transition: transform 0.2s ease-in-out, mask-image 0.2s ease-in-out;
+        }
+      }
+
+      &:hover,
+      &:focus {
+        &::before {
+          background-color: color(link--hover);
+          @if $animate == 'left' or $animate == 'true' {
+            transform: translateX(-0.2em);
+          }
```

Comment:
@cjwest I thought we agree on if icon is in "before" position, then there is no animation 😄 ? Also, I suggest we remove the left margin on the icon when it's in the before position, so when it's used in a button, the spacing looks more balanced on both sides. Currently, going from the screenshot, it looks like there is some extra spacing to the left of the email envelope icon.

Reply:
Good question, @yvonnetangsu.
I was thinking that for the mailto icon, there would be no animation. If we never want the animation for any icons in the "before" position, I'll remove it.

Reply:
Yeah, let's not allow any animation when the icon is in the before position. Thanks, @cjwest !